dubbo-common/src/main/java/org/apache/dubbo/rpc/executor/AbstractIsolationExecutorSupport.java

```diff
@@ -50,15 +50,17 @@ public Executor getExecutor(Object data) {
 
         for (URL serviceUrl : serviceUrls) {
             if (serviceUrl.getProtocol().equals(url.getProtocol()) && serviceUrl.getPort() == url.getPort()) {
-                return executorRepository.getExecutor(serviceUrl);
+                return executorRepository.getExecutor(providerModel, serviceUrl);
             }
         }
-        return executorRepository.getExecutor(serviceUrls.get(0));
+        return executorRepository.getExecutor(providerModel, serviceUrls.get(0));
     }
 
-    protected abstract ServiceKey getServiceKey(Object data);
+    protected ServiceKey getServiceKey(Object data) {
```

Comment:
why change this method from abstract ？

Reply:
Dubbo Protocol do not need to override this method.